### ISSUE TITLE
[Bug]:The scan feature does not work on a specific device

### ISSUE THREAD

**Issue by AdinaPetridean** [mozilla-mobile/fenix#27353]
### Steps to reproduce

1. On mobile, go to Settings - Synchronise and save your data - Ready to scan.
2. On Desktop sign in with a valid account.
3. On Desktop, open firefox.com/pair - Show code.
4. Try to scan the generated QR code with your device.

### Expected behaviour

The generated QR code is scanned, the user is redirected to the FxA sign in page.

### Actual behaviour

The camera doesn't read the generated QR code and the app is sent to background.

### Device name

Sony Xperia

### Android version

Android 6.0.1

### Firefox release type

Firefox Nightly

### Firefox version

Firefox Nightly 107.0a1 10/10 

### Additional information

Not reproducible on other devices.
Please note this issue is not reproducible via "Use email instead" option.
The scan feature does not work also for scanning a QR code to open a webpage via "Scan" button.

https://user-images.githubusercontent.com/108326843/194886624-7ce67579-d712-49b1-8ec2-6c39674f1919.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21770)

**Comment by Mugurell** [comment]
Did the app crashed?
Can you attach a logcat?

**Comment by AdinaPetridean** [comment]
@Mugurell 
I also thought it  was a crash, but the app is sent in the background.

**Comment by Mugurell** [comment]
@AdinaPetridean Can you add a logcat for when the issue happens?

**Comment by lobontiumira** [comment]
I was able to reproduce this issue on the latest Fenix Nightly build 107.0a1 from 10/11 with Samsung Galaxy Tab A6 (Android 5.1.1).
I managed to get a logcat: [logcat.txt](https://github.com/mozilla-mobile/fenix/files/9753655/logcat.txt)

**Comment by Mugurell** [comment]
In the logcat above I'm seeing a related error:
```
E/ExceptionHandler( 8975): java.lang.NoSuchMethodError: No virtual method sort(Ljava/util/Comparator;)V in class Ljava/util/ArrayList; or its super classes (declaration of 'java.util.ArrayList' appears in /system/framework/core-libart.jar)
E/ExceptionHandler( 8975):      at com.google.zxing.qrcode.detector.FinderPatternFinder$$ExternalSyntheticApiModelOutline0.m(R8$$SyntheticClass:3)
E/ExceptionHandler( 8975):      at com.google.zxing.qrcode.QRCodeReader.decode(QRCodeReader.java:544)
E/ExceptionHandler( 8975):      at com.google.zxing.MultiFormatReader.decodeInternal(MultiFormatReader.java:14)
E/ExceptionHandler( 8975):      at mozilla.components.feature.qr.QrFragment.decodeSource$feature_qr_release(QrFragment.kt:22)
E/ExceptionHandler( 8975):      at mozilla.components.feature.qr.QrFragment$imageAvailableListener$1$onImageAvailable$1.invokeSuspend(QrFragment.kt:23)
E/ExceptionHandler( 8975):      at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:9)
E/ExceptionHandler( 8975):      at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:107)
E/ExceptionHandler( 8975):      at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:82)
E/ExceptionHandler( 8975):      Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@15a2b869, Dispatchers.Default] 
```

@lobontiumira  @AdinaPetridean Does QR scanning work in other apps - eg: [Barcode Scanner](https://play.google.com/store/apps/details?id=com.google.zxing.client.android&hl=en&gl=US) ?

**Comment by lobontiumira** [comment]
@Mugurell yes, the Barcode Scanner app works on Samsung Galaxy Tab A6 (Android 5.1.1).

**Comment by AdinaPetridean** [comment]
@Mugurell also on Sony Xperia the Barcode Scanner app works.

**Comment by kbrosnan** [comment]
Duplicate bz-1804844

**Comment by gwarser** [comment]
Seems fixed in Nightly 110.0a1 (Build #2015922603.

Some joker set needinfo on Bugzilla and locked the bug.
